clientlibrary/worker/shard-consumer.go

```diff
@@ -151,25 +158,11 @@ func (sc *ShardConsumer) getRecords(shard *par.ShardStatus) error {
 	}
 	sc.recordProcessor.Initialize(input)
 
-	recordCheckpointer := NewRecordProcessorCheckpoint(shard, sc.checkpointer)
+	sc.recordCheckpointer = NewRecordProcessorCheckpoint(shard, sc.checkpointer)
 	retriedErrors := 0
 
 	for {
 		getRecordsStartTime := time.Now()
-		if time.Now().UTC().After(shard.LeaseTimeout.Add(-5 * time.Second)) {
```

Comment:
leases can already be expired by this point and claimed by someone else. this functionality has been moved to the `CheckpointSequence` function in checkpointer.